assets/src/components/searchResults.tsx

```diff
@@ -46,33 +23,36 @@ const SearchResultCard = ({
   onClick: (vehicle: VehicleOrGhost) => void
   isSelected: boolean
 }) => {
+  const { routeId, viaVariant, headsign } = vehicleOrGhost
+  const viaVariantFormatted = viaVariant && viaVariant !== "_" ? viaVariant : ""
+  const route = useRoute(vehicleOrGhost.routeId)
+
+  const title =
+    isVehicle(vehicleOrGhost) && vehicleOrGhost.isShuttle
+      ? "Shuttle"
+      : `${route?.name || routeId}_${viaVariantFormatted} ${headsign}`
+
```

Comment:
`RouteVariantName` encapsulates this behaviour I believe?

Maybe `Card.props.title` could be expanded to type `ReactElement` and then do the following? 
```
title={<RouteVariantName vehicle={vehicleOrGhost}/>}
```

Reply:
Though, If you don't want the `output` elements from `RouteVehicleName`(or it breaks tests), maybe they could be refactored out to a point where there's two functions which return strings as `ReactElements` which `RouteVehicleName` and other places could consume?

Reply:
Yeah I just didn't do it at first because of the type of `props.title` on `<Card />` but I just realized I can expand that to take an element as well without needing any further changes.

Reply:
Okay updated in 27e8febe.

Reply:
I see react prefer `ReactElement` over `JSX.Element`, I believe they may be aliases of each other? 
Should I prefer `JSX.Element` over `ReactElement` when creating new code?

Reply:
`JSX.Element` over `ReactElement` wasn't actually a particularly conscious decision on my part, I changed it in d257d0f0.